scrapy/core/engine.py

```diff
@@ -362,31 +402,30 @@ def _on_complete(_: _T) -> _T:
         dwld.addBoth(_on_complete)
         return dwld
 
-    @inlineCallbacks
-    def open_spider(
+    @deferred_f_from_coro_f
+    async def open_spider(
         self,
         spider: Spider,
-        start_requests: Iterable[Request] = (),
         close_if_idle: bool = True,
-    ) -> Generator[Deferred[Any], Any, None]:
-        if self.slot is not None:
+    ) -> None:
+        if self._slot is not None:
             raise RuntimeError(f"No free spider slot when opening {spider.name!r}")
         logger.info("Spider opened", extra={"spider": spider})
-        nextcall = CallLaterOnce(self._next_request)
+        self.spider = spider
+        nextcall = CallLaterOnce(self._start_scheduled_requests)
         scheduler = build_from_crawler(self.scheduler_cls, self.crawler)
-        start_requests = yield self.scraper.spidermw.process_start_requests(
-            start_requests, spider
+        self._slot = _Slot(close_if_idle, nextcall, scheduler)
+        self._start = await maybe_deferred_to_future(
+            self.scraper.spidermw.process_start(spider)
         )
-        self.slot = Slot(start_requests, close_if_idle, nextcall, scheduler)
-        self.spider = spider
```

Comment:
No technical reason for moving the definitions of spider and slot before the await, just felt cleaner.